src/panels/config/users/ha-dialog-add-user.js

```diff
@@ -0,0 +1,171 @@
+import '@polymer/paper-button/paper-button.js';
+import '@polymer/paper-dialog/paper-dialog.js';
+import '@polymer/paper-spinner/paper-spinner.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../../../components/ha-form.js';
+import '../../../components/ha-markdown.js';
+import '../../../resources/ha-style.js';
+
+import LocalizeMixin from '../../../mixins/localize-mixin.js';
+
+/*
+ * @appliesMixin LocalizeMixin
+ * @appliesMixin EventsMixin
+ */
+class HaDialogAddUser extends LocalizeMixin(PolymerElement) {
+  static get template() {
+    return html`
+    <style include="ha-style-dialog">
+      .error {
+        color: red;
+      }
+      paper-dialog {
+        max-width: 500px;
+      }
+      .username {
+        margin-top: -8px;
+      }
+    </style>
+    <paper-dialog id="dialog" with-backdrop opened="{{_opened}}" on-opened-changed="_openedChanged">
+      <h2>Add user</h2>
+      <div>
+        <template is="dom-if" if="[[_errorMsg]]">
+          <div class='error'>[[_errorMsg]]</div>
+        </template>
+        <paper-input
+          autofocus
+          class='username'
+          label='Username'
+          value='{{_username}}'
+          required
+          auto-validate
+          error-message='Required'
+        ></paper-input>
+        <paper-input
+          label='Password'
+          type='password'
+          value='{{_password}}'
+          required
+          auto-validate
+          error-message='Required'
+        ></paper-input>
+      </div>
+      <div class="buttons">
+        <template is="dom-if" if="[[_loading]]">
+        <div class='submit-spinner'><paper-spinner active></paper-spinner></div>
+        </template>
+        <template is="dom-if" if="[[!_loading]]">
+        <paper-button on-click="_createUser">Create</paper-button>
+        </template>
+      </div>
+    </paper-dialog>
+`;
+  }
+
+  static get properties() {
+    return {
+      _hass: Object,
+      _dialogClosedCallback: Function,
```

Comment:
if you want to use function you have to use deserialize

Reply:
deserialize is only used for things that are set via an attribute. This is an internal property. Alternative is that I just set it on `this`.